devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,365 @@
+
+**SRS_INTERLOCKED_43_012[** `interlocked_decrement_16` shall atomically decrement (decrease by one) the 16-bit variable `*addend`.**]**
+
+**SRS_INTERLOCKED_43_044: [** `interlocked_decrement_16` shall return the decremented value. **]**
+
+## interlocked_decrement_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_decrement_64, volatile int64_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_013[** `interlocked_decrement_64` shall atomically decrement (decrease by one) the 64-bit variable `*addend`.**]**
+
+**SRS_INTERLOCKED_43_045: [** `interlocked_decrement_64` shall return the decremented value. **]**
+
+
+## interlocked_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_exchange, volatile int32_t*, target, int32_t, value);
+```
+
+**SRS_INTERLOCKED_43_014 [** `interlocked_exchange` shall set the 32-bit variable pointed to by `target` to `value` as an atomic operation.**]**
+
+**SRS_INTERLOCKED_43_046: [** `interlocked_exchange` shall return the initial value pointed to by `target`. **]**
+
+## interlocked_exchange_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_exchange_16, volatile int16_t*, target, int16_t, value);
+```
+
+**SRS_INTERLOCKED_43_015 [** `interlocked_exchange_16` shall set the 16-bit variable pointed to by `target` to `value` as an atomic operation.**]**
+
+**SRS_INTERLOCKED_43_047: [** `interlocked_exchange_16` shall return the initial value pointed to by `target`. **]**
+
+## interlocked_exchange_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_exchange_64, volatile int64_t*, target, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_016 [** `interlocked_exchange_64` shall set the 64-bit variable pointed to by `target` to `value` as an atomic operation.**]**
+
+**SRS_INTERLOCKED_43_048: [** `interlocked_exchange_64` shall return the initial value pointed to by `target`. **]**
+
+## interlocked_exchange_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_exchange_8, volatile int8_t*, target, int8_t, value);
+```
+
+**SRS_INTERLOCKED_43_017 [** `interlocked_exchange_8` shall set the 8-bit variable pointed to by `target` to `value` as an atomic operation.**]**
+
+**SRS_INTERLOCKED_43_049: [** `interlocked_exchange_8` shall return the initial value pointed to by `target`. **]**
+
+## interlocked_exchange_add
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_exchange_add, volatile int32_t*, addend, int32_t, value);
+```
+
+**SRS_INTERLOCKED_43_018 [** `interlocked_exchange_add` shall perform an atomic addition of the 32-bit values `*addend` and `value` and store the result in `*addend`.**]**
+
+**SRS_INTERLOCKED_43_050: [** `interlocked_exchange_add` shall return the initial value of `*addend` **]**
+
+## interlocked_exchange_add_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_exchange_add_64, volatile int64_t*, addend, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_018 [** `interlocked_exchange_add_64` shall perform an atomic addition of the 64-bit values `*addend` and `value` and store the result in `*addend`.**]**
```

Comment:
>SRS_INTERLOCKED_43_018 [](start = 2, length = 22)

Your build was failing and it said there were duplicate requirements. It looks like this one `SRS_INTERLOCKED_43_018` and below `SRS_INTERLOCKED_43_050` are repeated. I would remove those tags from these two lines and re-tag it with the extension to get unique numbers.